docs/flow_filtering.md

```diff
@@ -0,0 +1,95 @@
+ 
+### Mandatory parameters
+
+- `FLOW_FILTER_IP_CIDR` - IP address and CIDR mask for the flow filter rule, supports IPv4 and IPv6 address format.
+  If wanted to match against any IP, user can use `0.0.0.0/0` or `::/0` for IPv4 and IPv6 respectively.
+- `FLOW_FILTER_ACTION` - Action to be taken for the flow filter rule. Possible values are `Accept` and `Reject`.
+  - For the matching rule with `Accept` action this flow will be allowed to be cached in eBPF table, with updated global metric `FlowFilterAcceptCounter`.
+  - For the matching rule with `Reject` action this flow will be dropped and not cached in eBPF table, with updated global metric `FlowFilterRejectCounter`.
+  - If the rule is not matched, the flow will be allowed to be cached in eBPF table, with updated global metric `FlowFilterNoMatchCounter`.
+
+### Optional parameters
+
+- `FLOW_FILTER_DIRECTION` - Direction of the flow filter rule. Possible values are `Ingress` and `Egress`.
+- `FLOW_FILTER_PROTOCOL` - Protocol of the flow filter rule. Possible values are `TCP`, `UDP`, `SCTP`, `ICMP`, `ICMPv6`.
+- `FLOW_FILTER_SOURCE_PORT` - Single Source port of the flow filter rule.
+- `FLOW_FILTER_SOURCE_PORT_RANGE` - Source port range of the flow filter rule. using "80-100" format.
+- `FLOW_FILTER_DESTINATION_PORT` - Single Destination port of the flow filter rule.
+- `FLOW_FILTER_DESTINATION_PORT_RANGE` - Destination port range of the flow filter rule. using "80-100" format.
+- `FLOW_FILTER_ICMP_TYPE` - ICMP type of the flow filter rule.
+- `FLOW_FILTER_ICMP_CODE` - ICMP code of the flow filter rule.
+- `FLOW_FILTER_IP` - Specific IP address of the flow filter rule.
+
+## How does it work
+
+### What does IP CIDR filtering work? 
+
+- When filtering ingress traffic, the source IP address is a crucial parameter used to match filter rules configured with CIDR notation.
+  CIDR notation allows for the efficient representation of IP address ranges by combining the base IP address with a prefix length
+  to denote the network subnet.
+- When filtering egress traffic, the destination IP address is a crucial parameter used to match filter rules configured with CIDR notation.
+  CIDR notation allows for the efficient representation of IP address ranges by combining the base IP address with a prefix length.
+
+### Matching specific endpoints using `FLOW_FILTER_IP`
+
+- When filtering ingress traffic, the `FLOW_FILTER_IP` will be used to match against the destination IP address of the packet, 
+  so after matching source IP CIDR now we can pinpoint to a specific endpoint via `FLOW_FILTER_IP` and based on the provisioned action,
+  flow with either be allowed to be cached in eBPF flow table or not.
+- When filtering egress traffic, the `FLOW_FILTER_IP` will be used to match against the source IP address of the packet,
+  so after matching destination IP CIDR now we can pinpoint to a specific endpoint via `FLOW_FILTER_IP` and based on the provisioned action,
```

Comment:
IMO the config names are quite confusing to represent what they're doing and I do need to remember what they'll match against depending upon the direction or would have to consult API doc, is it possible to make it more simpler? 
Also, `FLOW_FILTER_IP` doesn't accept CIDR, IIUC usually the cluster resources belong to fix CIDR space , it would be good to be able specify CIDR space for this config as well. I may not be aware any restriction/limitation that led to have specific IP instead of CIDR space.

Reply:
not sure why u need CIDRs for in cluster address as normally this is very specific IP?

Reply:
For `FLOW_FILTER_IP` config which would match in-cluster IPs, I was thinking about the use cases where I am interested in to see flows for multiple pods/service in a cluster NS to filter-in the flowlogs for.